lambdas/process-mobile-app-callback/src/main/java/uk/gov/di/ipv/core/processmobileappcallback/ProcessMobileAppCallbackHandler.java

```diff
@@ -0,0 +1,176 @@
+import uk.gov.di.ipv.core.library.exceptions.IpvSessionNotFoundException;
+import uk.gov.di.ipv.core.library.helpers.ApiGatewayResponseGenerator;
+import uk.gov.di.ipv.core.library.helpers.LogHelper;
+import uk.gov.di.ipv.core.library.helpers.RequestHelper;
+import uk.gov.di.ipv.core.library.persistence.item.CriResponseItem;
+import uk.gov.di.ipv.core.library.service.ClientOAuthSessionDetailsService;
+import uk.gov.di.ipv.core.library.service.ConfigService;
+import uk.gov.di.ipv.core.library.service.CriResponseService;
+import uk.gov.di.ipv.core.library.service.IpvSessionService;
+import uk.gov.di.ipv.core.library.service.exception.InvalidCriResponseException;
+import uk.gov.di.ipv.core.processmobileappcallback.dto.MobileAppCallbackRequest;
+import uk.gov.di.ipv.core.processmobileappcallback.exception.InvalidMobileAppCallbackRequestException;
+
+import java.util.Optional;
+
+import static uk.gov.di.ipv.core.library.journeyuris.JourneyUris.JOURNEY_ERROR_PATH;
+import static uk.gov.di.ipv.core.library.journeyuris.JourneyUris.JOURNEY_NEXT_PATH;
+
+public class ProcessMobileAppCallbackHandler
+        implements RequestHandler<APIGatewayProxyRequestEvent, APIGatewayProxyResponseEvent> {
+    private static final Logger LOGGER = LogManager.getLogger();
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+    private static final JourneyResponse JOURNEY_NEXT = new JourneyResponse(JOURNEY_NEXT_PATH);
+    private final ConfigService configService;
+    private final IpvSessionService ipvSessionService;
+    private final ClientOAuthSessionDetailsService clientOAuthSessionDetailsService;
+    private final CriResponseService criResponseService;
+
+    public ProcessMobileAppCallbackHandler(
+            ConfigService configService,
+            IpvSessionService ipvSessionService,
+            ClientOAuthSessionDetailsService clientOAuthSessionDetailsService,
+            CriResponseService criResponseService) {
+        this.configService = configService;
+        this.ipvSessionService = ipvSessionService;
+        this.clientOAuthSessionDetailsService = clientOAuthSessionDetailsService;
+        this.criResponseService = criResponseService;
+    }
+
+    @ExcludeFromGeneratedCoverageReport
+    public ProcessMobileAppCallbackHandler() {
+        configService = ConfigService.create();
+        ipvSessionService = new IpvSessionService(configService);
+        clientOAuthSessionDetailsService = new ClientOAuthSessionDetailsService(configService);
+        criResponseService = new CriResponseService(configService);
+    }
+
+    @Override
+    @Tracing
+    @Logging(clearState = true)
+    public APIGatewayProxyResponseEvent handleRequest(
+            APIGatewayProxyRequestEvent input, Context context) {
+        try {
+            var callbackRequest = parseCallbackRequest(input);
+
+            var journeyResponse = getJourneyResponse(callbackRequest);
```

Comment:
Might be clearer to just make this a void method `validateCallbackRequest` given this is what it's really doing, and then just set the default `JOURNEY_NEXT` response here?